packages/cli/src/commands/server/launchDebugger.js

```diff
@@ -77,7 +97,7 @@ function launchChrome(url: string) {
 }
 
 function launchDebugger(url: string) {
-  if (!commandExists(getChromeAppName())) {
+  if (!commandExists(getChromeCommandName())) {
```

Comment:
I think we could get rid of `commandExists`, `getChromeCommandName` and `getChromeAppName` if we try/catch the `launchChrome(url)` and then fall back to `launchDefaultBrowser(url)` with the info log. What do you think?

Reply:
@thymikee That is what I thought could be done here, but, for some reason, when I played with the code, I could not manage the `open` library to get to error handling code when I passed some garbage command to it. It seemed it just swallowed any errors... Maybe I will give it another try.

Reply:
Use async/await interface of open, that should help

Reply:
Do you mean something like that?

```
try {
    launchChrome(url);
  } catch (error) {
    launchDefaultBrowser(url);
	  logger.info(
      `For a better debugging experience please install Google Chrome from: 	 
       ${chalk.underline.dim('https://www.google.com/chrome/',
      )}`,);
  }
```

Reply:
You're missing the key part: `await launchChrome` and converting launchChrome into an `async` function

Reply:
Yes sure, but we will need `getChromeAppName` anyways because `launchChrome` used it, right?

Reply:
if it uses it then sure, let it stay

Reply:
looking at `open`'s code gave me some clue. You also need to pass wait: true, for it to pass any errors...
So, it becomes like this:

```
function launchChrome(url) {
  return (0, _open().default)(url, {
    app: [getChromeAppName()],
    wait: true,
  });
}

async function launchDebugger(url) {
  try {
    await launchChrome(url);
  } catch (err) {
    _cliTools().logger.info(`For a better debugging experience please install Google Chrome from: ${_chalk().default.underline.dim('https://www.google.com/chrome/')}`);
    (0, _launchDefaultBrowser.default)(url);
  }
}
```
also need to delete commandExists and commandExistsWindowsSync functions.

Reply:
Submitted an alternative PR here: https://github.com/react-native-community/cli/pull/771 :)